### PR TITLE
TERBS DAY OF FUN 1: Soda

### PR DESCRIPTION
**Pull request by Iamgoofball**
Sorry about the map thing, mapmerge doesn't work on ubuntu.

This PR adds Terbus Soda to the game. Very unhealthy soda can now be drunk by crewmembers! Fully emulate SpaceDonalds!

### CODE REVIEW

**Comment by Zelacks** [comment]
You should definetely get someone to map merge for you. I am pretty sure there are a few reasons that map merger is used, one being that the large amount of lines added and removed largely adds to the time that it takes github to do operations. (maybe? I am probably just spreading misconceptions)
It also is harder to tell what you have added and removed from the map, as git wont tell us, who knows perhaps you even put several eswords onto the map.
also probably a side note is that it screws up contribution statistics

**Comment by Zelacks** [comment]
Also a 2048 volume beaker is totally abusable. splashing someone would be a death sentence.
I was thinking about filling the beaker with water and splashing myself with thewater to get to the reagent cap, but since ergo changed how water works, splashing water doesnt actually transfer the reagent.

File: NTstation13.int [hunk 1]
```diff
@@ -1,6 +1,11 @@
 // BEGIN_INTERNALS
 /*
 MAP_ICON_TYPE: 0
+WINDOW: code\modules\reagents\Chemistry-Machinery.dm;code\modules\reagents\Chemistry-Reagents.dm;code\modules\reagents\reagent_containers\glass.dm;icons\obj\chemical.dmi;icons\obj\drinks.dmi;icons\obj\reagentfillings.dmi;icons\obj\food.dmi
```
Comment (RemieRichards) [inline NTstation13.int:4]: STOP COMMITTING THIS DAMN FILE.

**Comment by Ccomp5950** [comment]
"Sorry about the map thing, mapmerge doesn't work on ubuntu."

java -jar MapPatcher.jar

You might learn something...

**Comment by Atskadan** [comment]
>committing the .dme
>no map merge
>direct references to terbs with no actual fun easter eggs or anything

gooby pls
there's nothing wrong with references to players, but IMO they should be reserved for small easter eggs or plaques for infamous players or events in /tg/ history, like the meteor plaque or doublebearded wine. "Now with Dann-O's!" is a good one as well.
outright naming the machine and all of its drinks after the person who suggested the idea isnt tasteful

**Comment by Iamgoofball** [comment]
@ccomp5950 I did that, but from what it looks like git decided "oh you mapmerged? Well fuck you I'm changing the entire file anyways"

@atskadan Alright, I'll change thenames.

@Zelacks I'll switch it over to a drink subtype so you can't splash.

**Comment by Ccomp5950** [comment]
The mapmerge works in ubuntu, it works in BSD, it works in Haiku (BeOS), you know why it works?  Because it's java.  You just don't have the batch files

You know what, fuck it.

```
 (be in the root directory of the project, where you could cd code to view code or cd maps to look at maps)
 git show HEAD~9:maps/NTstation13.dmm > maps/NTstation13.backup
 java -jar tools/mapmerge/MapPatcher.jar -clean maps/NTstation13.backup maps/NTstation13.dmm maps/NTstation13.dmm
 git add maps/NTstation13.dmm
 git commit -m "I'm a derp"
 git push origin traitor
```

There your map is fixed.

We use HEAD~9 so it pulls the map from 9 commits back (I'm looking at your PR being 8 commits right now) and makes a file from it called NTStation13.backup

you then use the MapPatcher.jar to clean the map for you.
MapPatcher.jar -clean expects it's syntax to be as so...
java -jar MapPatcher.jar -clean [OLD MAP] [NEW MAP] [WHERE DO I SAVE IT TO]
so we use the old map that we saved as NTstation13.backup as the first argument, your current map as new map, and we just save over the new map with the cleaned map by specifying it again.

How you can save yourself trouble in the future

before you make any map changes...

```
 cp maps/NTstation13.dmm maps/NTstation13.backup
 <make your map changes>
 java -jar tools/mapmerge/MapPatcher.jar -clean maps/NTstation13.backup maps/NTstation13.dmm maps/NTstation13.dmm
```

Done, you just did manually what prepare.bat and make_clean.bat does for you.

Troubleshooting:

```
 It says I don't have java.
```

WELL GET JAVA

```
 it says file doesn't exist
```

YOU ARE IN THE WRONG DIRECTORY

```
 it didn't work, obviously mappatcher doesn't work in <whatever os I'm using>
```

PEBKAC

**Comment by Ccomp5950** [comment]
and while you are at it.

```
 git show HEAD~9:NTstation13.int > NTstation13.int
 git add NTStation13.int
 git commit -m "I'll never commit this again, promise"
 git push origin traitor
 git update-index --assume-unchanged NTstation13.int
```

That last command, means you will never commit it again.  Because git will assumed it's never changed so unless you add it to your commit manually using git add NTstation13.int it won't be added and you can continue doing lazy...

```
 git add .
```

all day long without worrying about commiting it every again.

**Comment by ghost** [comment]
You can make bombcap hand grenades with this...

**Comment by Atskadan** [comment]
why not make the sodas way smaller but force the reagent transfer to be 1 unit per sip so its like youre drinking a lot more, when you arent really, thus removing the issues outline above
the max size should probably 50 so you cant get free Better Than Beakers™ Brand Sodas

**Comment by Iamgoofball** [comment]
what I was gonna do was turn them into reagent_containers/snacks/drinks/terbus so that I can set up a var in the dispensers to not let them be put into the standard chemical dispensers or grenades or p. much anything else w/ beaker related activities, but they CAN be put into the soda dispenser

this way we can have child size terbus but without bombcap terbus

**Comment by Atskadan** [comment]
that sounds kind of snowflaky tho
"oh yeah theyre reagent containers, except, they're specifically exempt from being reagent containers in every way, except they can be put in a soda dispenser and hold soda"

**Comment by Iamgoofball** [comment]
not really, I'll show you my implementation of it soon once I can clone this repo properly

**Pull request closed without merging**